### ISSUE TITLE
CardBook (and other clients) are only able to choose vCard 3.0 not 4.0 (which is the correct one)

### ISSUE THREAD

**Issue by tasn** [etesync/etesync-dav#66]
As mentioned in #41

I used to be able to choose 4.0, so I'm not sure what happened. Though it's probably something we (or radicale) are doing wrong. Needs further investigation.

**Comment by paapereira** [comment]
@tasn I'm also having the same issue, and having troubles syncing between Thunderbird Cardbook addon on Arch Linux and Etesync Android client.
If I can help debugging, testing.. let me know.

**Comment by tasn** [comment]
Apparently the reason why I used to be able to choose 4.0 was a bug in CardBook that has since been fixed. It looks like Radicale doesn't correctly advertise vCard 4.0 support (https://github.com/Kozea/Radicale/issues/832) although for all I know, it should support it just fine.

There's some information and more links in the radicale issue I linked to above (https://github.com/Kozea/Radicale/issues/832). Just posting it here for reference.

This also seems relevant for #65

**Comment by paapereira** [comment]
Don't know if it helps but I noted the following:

- Adding etesync account in Cardbook only has version 3.0 available
- If I add a new contact and sync it to Android I can see in the journal, in the RAW view, that the VCARD version is 3.0
- If I add an email address (for example) in Android in that contact (via Google Contacts) and sync it I can see in the journal that the version is now 4.0
- Syncing again in Thunderbird and the contact has different fields (example the Gender filed is now a dropdown)

The problems start when I add a picture in Thunderbird and then change something in Android. The birthday format is also giving me problems.

Bottom line, assuming Cardbook as support for 4.0 (never had these issues connecting to GMail contacts) then we need to make the 4.0 option available in Cardbook when adding the etesync account.

Bought my yearly subscription last Tuesday :P

**Comment by tasn** [comment]
The problem/reason for this back and forth from v3 and v4 is that the EteSync Android app is forcing the usage of v4, no matter what, and CardBook is forcing v3. They are mostly compatible (and the Android app supports both anyway) so both are able to read and write eachother's but then write in their own preferred format.

One fix that I'd really rather not do is change the Android (and web) app to use v3. What I'd much rather prefer though to add v4 support to Radicale.

**Comment by tasn** [comment]
I've done a bit more research and changing Android to v3 is unfortunately not an option. As I remembered, we also use the KIND:group vCard v4 feature which means groups are saved in separate vCards rather than as labels/orgs, so changing to v3 is not a simple flick of a setting. Only way forward then is to implement it in Radicale.

**Comment by tasn** [comment]
Just a random thought: maybe we don't need to actually implement v4 support in the underlying library (vobject), but rather just patch Radicale to fake report v4 support. Radicale only lightly uses vobject and where it does (uid) v3 and v4 are compatible. Just a thought worth exploring.

**Comment by stevesbrain** [comment]
In [this issue](https://gitlab.com/CardBook/CardBook/issues/634) there is an XPI that uses v4; but etesync-dav still doesn't include the images so limited contacts are able to be synced with Cardbook from there.

**Comment by tasn** [comment]
I replied in #65. It's definitely an issue with EteSync-dav (radicale).
